docs/tutorial/CustomDataLoaderTF.ipynb

```diff
@@ -1,573 +1,573 @@
+    "print('The version of SecretFlow: {}'.format(sf.__version__))\n",
+    "\n",
+    "# In case you have a running secretflow runtime already.\n",
+    "sf.shutdown()\n",
+    "sf.init(['alice', 'bob', 'charlie'], address=\"local\", log_to_driver=False)\n",
+    "alice, bob ,charlie = sf.PYU('alice'), sf.PYU('bob') , sf.PYU('charlie')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "25c38ba4",
+   "metadata": {},
+   "source": [
+    "## Interface Introduction"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "509e5461",
+   "metadata": {},
+   "source": [
+    "We support custom DataBuilder reads in SecretFlow's `FLModel` to make it easier for users to handle data inputs more flexibly according to their needs.\n",
+    "Let's use an example to demonstrate how to use the custom DataBuilder for federated model training.\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "ff157baf",
+   "metadata": {},
+   "source": [
+    "Steps to use DataBuilder:\n",
+    "1. Use the standalone version engine (TensorFlow, PyTorch) to develop and get the Dataset's Builder function\n",
+    "2. Wrap the Builder functions of each party to get `create_dataset_builder` function. *Note: The dataset_builder needs to pass in the stage parameter.*\n",
+    "3. Build the data_builder_dict [PYU, dataset_builder].\n",
+    "4. Pass the obtained data_builder_dict to the `dataset_builder` of the `fit` function. At the same time, the x parameter position is passed into the required input in dataset_builder. (eg: the input passed in this example is the actual image path used)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "fad75fd6",
+   "metadata": {},
+   "source": [
+    "Using DataBuilder in `FLModel` requires a pre-defined `data_builder_dict`. Need to be able to return `tf.dataset` and `steps_per_epoch`. And the steps_per_epoch returned by all parties must be consistent.\n",
+    "```python\n",
+    "data_builder_dict = \n",
+    "        {\n",
+    "            alice: create_alice_dataset_builder(\n",
+    "                batch_size=32,\n",
+    "            ), # create_alice_dataset_builder must return (Dataset, steps_per_epoch)\n",
+    "            bob: create_bob_dataset_builder(\n",
+    "                batch_size=32,\n",
+    "            ), # create_bob_dataset_builder must return (Dataset, steps_per_epochstep_per_epochs)\n",
+    "        }\n",
+    "\n",
+    "```"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "c55b0faf",
+   "metadata": {},
+   "source": [
+    "## Download Data"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "07ffc09e",
+   "metadata": {},
+   "source": [
+    "Flower Dataset Introduction: The Flower dataset consists of 4323 color images of 5 different types of flowers (daisy, dandelion, rose, sunflower, and tulip). Each flower category has multiple angles and different lighting conditions. The resolution of each image is 320x240 pixels.\n",

```

Comment:
Each flower category has multiple angles and different lighting conditions.

Please retranaslate the sentence.

Reply:
Each flower has images from multiple angles and different lighting conditions, and the resolution of each image is 320x240.